app/generators/questionnaire_generator.rb

```diff
@@ -8,10 +8,17 @@ def initialize
   end
 
   # rubocop:disable Metrics/ParameterLists
-  def generate_questionnaire(response_id:, content:, title:, submit_text:, action:, unsubscribe_url:, params: {})
+  # rubocop:disable Metrics/AbcSize
+  def generate_questionnaire(response_id:, content:, title:, submit_text:,
+                             action:, unsubscribe_url:, locale:, params: {})
     params[:response_id] = response_id
+    params['content[locale]'] = locale
     response = Response.find_by(id: response_id) # allow nil response id for preview
-    raw_content = content[:questions].deep_dup
+    raw_content = QuestionnaireTranslator.translate_content(content[:questions].deep_dup, 'i18n')
```

Comment:
so essentially we don't translate it if it has an i18n attribute, right? We just use that key.

Reply:
We do translate it, we translate it to the i18n language.